packages/react/src/jsx.tsx

```diff
@@ -1,16 +1,18 @@
 import type { JSXElementConstructor, ReactElement } from 'react'
 
-import { event, ServerEventDataState } from './events'
+import { event, ServerDataState } from './events'
 import {
   ActionDataState,
   ComponentDataState,
   DataState,
   isComponentDataState,
   JSONValue,
   ReferencedDataState,
+  ServerEventDataState,
   StateDataState,
 } from './template'
 
+// tbd: refactor and unify this
 type AllowedDataStates =
```

Comment:
This should not be needed, but should match `ComponentDataState['props']`. Right now, ComponentDataState is more relaxed, and this is stricter, so we need to unify. Not a big deal for now, but needs to be done.